contracts/interfaces/IAxelarGMPExecutable.sol

```diff
@@ -24,7 +24,7 @@ interface IAxelarGMPExecutable {
      * @notice Returns the address of the AxelarGMPGateway contract.
      * @return The Axelar GMP Gateway contract associated with this executable contract.
      */
-    function gateway() external view returns (IAxelarGMPGateway);
+    function gatewayAddress() external view returns (address);
```

Comment:
Forgot this was public, would prefer to not break this interface
```suggestion
    function gateway() external view returns (IAxelarGMPGateway);
```
maybe fine to just have this by default, `gatewayWithToken` can be internal if someone needs it. For Amplifier chains, the gateway is going to be pure messaging anyways.